R-Interface/jaspResults/src/jaspPlot.cpp

```diff
@@ -57,17 +57,55 @@ void jaspPlot::initEnvName()
 void jaspPlot::setPlotObject(Rcpp::RObject obj)
 {
 	Rcpp::List plotInfo = Rcpp::List::create(Rcpp::_["obj"] = obj, Rcpp::_["width"] = _width, Rcpp::_["height"] = _height, Rcpp::_["revision"] = _revision);
-	_filePathPng = "";
+
+	if (!_editing)
+		_filePathPng = "";
+
+	jaspResults::setObjectInEnv(_envName, plotInfo);
+
+	if (connectedToJaspResults())
+		renderPlot();
+
+}
+
+void jaspPlot::renderPlot()
+{
+	// if a png exists the plot was already rendered, unless we're editing it
+	if (_filePathPng != "" && !_editing)
+		return;
+
+	// empty plots were added to the state
+	Rcpp::RObject plotInfoObj = jaspResults::getObjectFromEnv(_envName);
+	if (plotInfoObj.isNULL())
+		return;
+
+	Rcpp::List plotInfo = Rcpp::as<Rcpp::List>(plotInfoObj);
+	Rcpp::RObject obj = plotInfo["obj"];
 
 	if(!obj.isNULL())
 	{
+
+		jaspPrint("Now rendering a plot!");
```

Comment:
It might be helpful to add the name or title to this debugstatement, that way if in the future we get logs from a user where the plotrendering failed we can pinpoint in the logs which plot is breaking it.
Because we can no longer assume the order in which the functions are called are the same order the plots are rendered.

Reply:
It certainly *is* helpful to havve a printstatement here btw!